internal/provider/models/processors/trace_sampling.go

```diff
@@ -0,0 +1,250 @@
+						Computed:    true,
+						Optional:    true,
+						Description: "The generated name of the rule.",
+					},
+				}),
+			},
+		},
+	}),
+}
+
+func TraceSamplingProcessorFromModel(plan *TraceSamplingProcessorModel, previousState *TraceSamplingProcessorModel) (*client.Processor, diag.Diagnostics) {
+	dd := diag.Diagnostics{}
+
+	// a lot of config props are marked optional, so we need to check if they are set
+	// because they may not actually be optional or applicable given the type
+	if !ValidateRequiredConfigsPerSampleType(plan, plan.SampleType.ValueString(), &dd) {
+		return nil, dd
+	}
+
+	component := client.Processor{
+		BaseNode: client.BaseNode{
+			Type:        TRACE_SAMPLING_PROCESSOR_NODE_NAME,
+			Title:       plan.Title.ValueString(),
+			Description: plan.Description.ValueString(),
+		},
+	}
+	if previousState != nil {
+		component.Id = previousState.Id.ValueString()
+		component.GenerationId = previousState.GenerationId.ValueInt64()
+	}
+	component.Inputs = modelutils.StringListValueToStringSlice(plan.Inputs)
+	component.UserConfig = map[string]any{
+		"sample_type": plan.SampleType.ValueString(),
+	}
+	if !plan.TraceIdField.IsNull() && plan.TraceIdField.ValueString() != "" {
+		component.UserConfig["trace_id_field"] = plan.TraceIdField.ValueString()
+	}
+
+	if plan.SampleType.ValueString() == "head" {
+		if !plan.Rate.IsNull() && plan.Rate.ValueInt64() != 0 {
+			component.UserConfig["rate"] = plan.Rate.ValueInt64()
+		}
+	} else {
+		if !plan.ParentSpanIdField.IsNull() && plan.ParentSpanIdField.ValueString() != "" {
+			component.UserConfig["parent_span_id_field"] = plan.ParentSpanIdField.ValueString()
+		}
+
+		var conditionals []map[string]any
+		for _, entry := range plan.Conditionals.Elements() {
+			conditionals = append(conditionals, map[string]any{
+				"conditional":  modelutils.UnwindConditionalFromModel(entry.(basetypes.ObjectValue).Attributes()["conditional"]),
+				"rate":         entry.(basetypes.ObjectValue).Attributes()["rate"].(basetypes.Int64Value).ValueInt64(),
+				"_output_name": entry.(basetypes.ObjectValue).Attributes()["_output_name"].(basetypes.StringValue).ValueString(),
+			})
+		}
+
+		component.UserConfig["conditionals"] = conditionals
+	}
+
+	return &component, dd
+}
+
+func TraceSamplingProcessorToModel(plan *TraceSamplingProcessorModel, component *client.Processor) {
+	plan.Id = types.StringValue(component.Id)
+	if component.Title != "" {
+		plan.Title = types.StringValue(component.Title)
+	}
+	if component.Description != "" {
+		plan.Description = types.StringValue(component.Description)
+	}
+	plan.GenerationId = types.Int64Value(component.GenerationId)
+	plan.Inputs = modelutils.SliceToStringListValue(component.Inputs)
+	plan.SampleType = types.StringValue(component.UserConfig["sample_type"].(string))
+	plan.TraceIdField = types.StringValue(component.UserConfig["trace_id_field"].(string))
+
+	if rate, ok := component.UserConfig["rate"].(float64); ok {
```

Comment:
Should we check the sample type before setting rate and conditionals?
We've been burned in the past when the API suddenly started setting values which TF wasn't expecting